docs/static_site/src/pages/get_started/osx_setup.md

```diff
@@ -24,176 +24,151 @@ permalink: /get_started/osx_setup
+Building MXNet from source is a two-step process:
 
-**Note:** To change the compilation options for your build, edit the ```make/config.mk``` file and submit a build request with the ```make``` command.
+1. Build the shared library from the MXNet C++ source code.
+2. (optional) Install the supported language-specific packages for MXNet.
 
-## Prepare Environment for GPU Installation
+If you plan to build with GPU, you need to set up the environment for CUDA and
+cuDNN. Please follow the [NVIDIA CUDA Installation Guide for Mac OS
+X](https://docs.nvidia.com/cuda/cuda-installation-guide-mac-os-x/index.html) and
+[cuDNN Installation
+Guide](https://docs.nvidia.com/deeplearning/sdk/cudnn-install/index.html#install-mac).
+Note that CUDA stopped supporting macOS in 2019 and future versions of CUDA may
+not support macOS.
 
-This section is optional. Skip to next section if you don't plan to use GPUs. If you plan to build with GPU, you need to set up the environment for CUDA and cuDNN.
+## Contents
 
-First, download and install [CUDA 8 toolkit](https://developer.nvidia.com/cuda-toolkit).
+* [Build the MXNet shared library from source](#build-mxnet-from-source)
+* [Install Language Packages](#installing-language-packages-for-mxnet)
+    * [R](#install-the-mxnet-package-for-r)
+    * [Julia](#install-the-mxnet-package-for-julia)
+    * [Scala](#install-the-mxnet-package-for-scala)
+    * [Java](#install-the-mxnet-package-for-java)
+    * [Perl](#install-the-mxnet-package-for-perl)
+  * [Contributions](#contributions)
+  * [Next Steps](#next-steps)
 
-Once you have the CUDA Toolkit installed you will need to set up the required environment variables by adding the following to your ~/.bash_profile file:
+<hr>
 
-```bash
-    export CUDA_HOME=/usr/local/cuda
-    export DYLD_LIBRARY_PATH="$CUDA_HOME/lib:$DYLD_LIBRARY_PATH"
-    export PATH="$CUDA_HOME/bin:$PATH"
-```
 
-Reload ~/.bash_profile file and install dependencies:
-```bash
-    . ~/.bash_profile
-    brew install coreutils
-    brew tap caskroom/cask
-```
+## Build the MXNet shared library from source
 
-Then download [cuDNN 5](https://developer.nvidia.com/cudnn).
+On OS X, you need the following dependencies:
 
-Unzip the file and change to the cudnn root directory. Move the header files and libraries to your local CUDA Toolkit folder:
+**Step 1:** Install prerequisite packages.
 
 ```bash
-    $ sudo mv include/cudnn.h /Developer/NVIDIA/CUDA-8.0/include/
-    $ sudo mv lib/libcudnn* /Developer/NVIDIA/CUDA-8.0/lib
-    $ sudo ln -s /Developer/NVIDIA/CUDA-8.0/lib/libcudnn* /usr/local/cuda/lib/
-```
+# Install OS X Developer Tools
+xcode-select --install
+
+# Install Homebrew
+/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
 
-Now we can start to build MXNet.
+# Install dependencies
+brew install cmake ninja ccache opencv
+```
 
-## Build the Shared Library
+`opencv` is an optional dependency. You can delete it from above `brew install`
```

Comment:
It might be clearer to explain what's the consequence if `opencv` is not installed so users can make an informed decision.